docs/HOL_BDD-Security-Framework-integration-into-VSTS/README.md

```diff
@@ -0,0 +1,317 @@
+
+    ![](media/29.png)
+
+
+
+### Task 4: Update BDD-Security to include more tests
+
+**Step 1.** Lets add more tests to be run by BDD-Security. Navigate to the location where you keep your PartsUnlimitedMRP repository and open **SSH-MRP-BDD-Security.ps1** script located at **PartsUnlimitedMRP -> deploy**.
+
+
+**Step 2.** Replace `./gradlew -Dcucumber.options="--tags @app_scan --tags ~@skip"` with:
+
+ ```
+ ./gradlew -Dcucumber.options="--tags @app_scan,@host_config --tags ~@skip"
+ ```
+
+
+**Step 3.** At this stage you should also address a possibility of insecure website being promoted to production environment. Add the following code at the end of the script to fail the release if there were errors in test results:
+```
+# If file with results contains "failed" then returning non-zero value will indicate error and fail the build.
+return @(Get-Content all_tests.xml | Where-Object { $_.Contains("failed") }).Count
+```
+>**Note:** This is not required to fail the build if you have "Fail on Standard Error" ticked (by default) on "BDD-Security" task.
+    ![](media/30.png)
+
+
+**Step 4.** Save `SSH-MRP-BDD-Security.ps1`. Open command line (which supports git) and navigate to your PartsUnlimitedMRP repository location. Run the following commands to add these changes to the remote repo and trigger automatic build and release.
+
+```
+git add .
+git commit -m "added more tests to run by SSH-MRP-BDD-Security.ps1"
+git push
+```
+
+
+**Step 5.** Once the build and release have completed executing, open test runs in VSTS again and click on the latest result.
+
+  ![](media/31.png)
+
+
+**Step 6.** You should see the summary indicating that one test has failed. To investigate the reason of failure, download the attachment `all_tests.xml` by clicking on it.
+
+  ![](media/32.png)
+
+
+**Step 7.** Open downloaded `all_tests.xml` in text editor and find which test has failed. Based on the stack trace, a port test has failed by finding open ports other than 80 and 443. To fix this issue, you will need to modify the test to expect 22 to be open and run tests again.
+
+  ![](media/33.png)
+
+>**Note:** Do not close port 22 on the Ubuntu VM. It's the port used by SSH to connect to the machine.
+
+
+**Step 8.** Run the following command on the Ubuntu machine to open file with the failing test in nano editor:
+```
+nano bdd-security/src/test/resources/features/host_config.feature
+```
+
+
+**Step 9.** Add port 22 to the list of expected open ports like so:
+
+   ![](media/34.png)
+
+
+**Step 10.** Save the changes and exit the editor. Press `Ctrl+x` to indicate that you want to exit the editor. Press `Y` to save the changes and press `Enter` to confirm the file name.
+
+
+**Step 11.** Rerun the tests by triggering a new deployment or by running the following commands on the Ubuntu machine to see whether the tests will pass this time: 
+```
+cd bdd-security/
+./gradlew -Dcucumber.options="--tags @app_scan,@host_config --tags ~@skip"
+```
```

Comment:
The test is still failing, could be great if we can have a successful one at the end.